tests/queries/0_stateless/02302_column_decl_null_before_defaul_value.sql

```diff
@@ -0,0 +1,51 @@
+select 'create table, column +type +NULL';
+DROP TABLE IF EXISTS null_before SYNC;
+CREATE TABLE null_before (id INT NULL) ENGINE=MergeTree() ORDER BY tuple();
+DESCRIBE TABLE null_before;
+
+select 'create table, column +type +NOT NULL';
+DROP TABLE IF EXISTS null_before SYNC;
+CREATE TABLE null_before (id INT NOT NULL) ENGINE=MergeTree() ORDER BY tuple();
+DESCRIBE TABLE null_before;
+
+select 'create table, column +type +NULL +DEFAULT';
+DROP TABLE IF EXISTS null_before SYNC;
+CREATE TABLE null_before (id INT NULL DEFAULT 1) ENGINE=MergeTree() ORDER BY tuple();
+DESCRIBE TABLE null_before;
+
+select 'create table, column +type +NOT NULL +DEFAULT';
+DROP TABLE IF EXISTS null_before SYNC;
+CREATE TABLE null_before (id INT NOT NULL DEFAULT 1) ENGINE=MergeTree() ORDER BY tuple();
+DESCRIBE TABLE null_before;
+
+select 'create table, column +type +DEFAULT +NULL';
+DROP TABLE IF EXISTS null_before SYNC;
+CREATE TABLE null_before (id INT DEFAULT 1 NULL) ENGINE=MergeTree() ORDER BY tuple();
+DESCRIBE TABLE null_before;
+
+select 'create table, column +type +DEFAULT +NOT NULL';
+DROP TABLE IF EXISTS null_before SYNC;
+CREATE TABLE null_before (id INT DEFAULT 1 NOT NULL) ENGINE=MergeTree() ORDER BY tuple();
+DESCRIBE TABLE null_before;
+
+select 'create table, column -type +NULL +DEFAULT';
+DROP TABLE IF EXISTS null_before SYNC;
+CREATE TABLE null_before (id NULL DEFAULT 1) ENGINE=MergeTree() ORDER BY tuple();
```

Comment:
Uh, do we really want to allow column specifications without type?

Reply:
My observation is that documentation and implementation regarding column declaration are contradictory. See #37087, also considerations in #37229. I'd rather consider what make sense and execute correspondingly

@tavplubix I'd appreciate for an opinion

Reply:
Column specifications without explicit type still make sense

Reply:
Ah, the documentation says so already: "If the default expression is defined, the column type is optional. If there isn’t an explicitly defined type, the default expression type is used." Okay then.

Reply:
BTW, it doesn't work currently on master:
```sql
:) CREATE TABLE null_before (id NULL DEFAULT 1) ENGINE=MergeTree() ORDER BY tuple()

CREATE TABLE null_before
(
    `id` NULL DEFAULT 1
)
ENGINE = MergeTree
ORDER BY tuple()

Received exception from server (version 22.6.1):
Code: 50. DB::Exception: Received from localhost:9000. DB::Exception: Unknown data type family: NULL. (UNKNOWN_TYPE)
```